### ISSUE TITLE
Running on Heroku

### ISSUE THREAD

**Issue by staringispolite** [muffinista/chatterbot#37]
Any experience running this on Heroku? I'm getting the following, which doesn't happen on my local environment. It's almost like the working directory or path isn't set right. I'm new to Ruby so I'm having difficulty debugging.

2015-10-07T07:12:01.854202+00:00 heroku[slug-compiler]: Slug compilation started
2015-10-07T07:12:01.854211+00:00 heroku[slug-compiler]: Slug compilation finished
2015-10-07T07:12:02.179574+00:00 heroku[worker.1]: State changed from crashed to starting
2015-10-07T07:12:05.156281+00:00 heroku[worker.1]: Starting process with command `ruby suphypebot.rb`
2015-10-07T07:12:05.836093+00:00 heroku[worker.1]: State changed from starting to up
2015-10-07T07:12:07.067031+00:00 app[worker.1]:     from suphypebot.rb:4:in `<main>'
2015-10-07T07:12:07.067009+00:00 app[worker.1]:     from /app/vendor/ruby-2.0.0/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in`require'
2015-10-07T07:12:07.066962+00:00 app[worker.1]: /app/vendor/ruby-2.0.0/lib/ruby/2.0.0/rubygems/core_ext/kernel_require.rb:55:in `require': cannot load such file -- chatterbot/dsl (LoadError)
2015-10-07T07:12:07.882005+00:00 heroku[worker.1]: State changed from up to crashed
2015-10-07T07:12:07.870906+00:00 heroku[worker.1]: Process exited with status 1

Also, I used the ClearDB addon for mysql, and Heroku's config vars to set the corresponding db_uri. Not sure if that's correct - can't test the DB connection until this is fixed

**Comment by staringispolite** [comment]
Oops, this particular error was because of an incorrect Procfile. Should say something like

> worker: bundle exec ruby mybot.rb

However, the script is still trying to configure itself, presumably because it's not connecting to the DB. Any advice on the best way to deploy to heroku would be helpful. I'm thinking my options are:

* Change the config loading logic to look for db_uri in the Heroku config vars
* Commit my YML file to a public repo (...dangerous to say the least)

**Comment by muffinista** [comment]
FYI, the beta version of the gem doesn't support MySQL at all. For 1.0, can you specify environment variables for a Heroku instance? I think you can. If so, you could run the bot with something like:

```
CHATTERBOT_DB="mysql://foo" bundle exec ruby mybot.rb
```

I'm contemplating ways to add SQL support back, but it was a drag to support and made the configuration inflexible.

**Comment by staringispolite** [comment]
Oh interesting. I just forked master here, and got it working fine off mysql locally: https://github.com/staringispolite/TwitterHypeMan

Starting to think forking master may have been the wrong way to go about things? :) 

(And yes, Heroku lets you add key/value pairs to the environment) Ruby gets at them via ENV[key]

**Comment by muffinista** [comment]
It depends on exactly what you need to do. If you're running your bot via the Streaming API, then you don't need to persist anything, and it should be fine running on Heroku. If you want to run it as a cron job, or if there's any data you need to store, then the master branch might not work too well for you.

I have a thought about how to handle the persistence problem, I'll see if I can get some code going in the next few days.

**Comment by staringispolite** [comment]
Cool! I ended up switching to the streaming API for the reasons you said just to get a working version up (great minds!)

@alexbaldwin and I added some config stuff for local vs heroku environments, so that I wouldn't have to push my Twitter key/secret/etc into my public git repo - happy to send a PR if you're interested. 

I only wanted persistence so it wouldn't miss an @-mention. (Say, if I have downtime, or if Heroku sleeps the dyno). It seems like the streaming API would only send it @-mentions as they happen, so if it's down when a tweet happens, it'll never process that tweet. Not the biggest deal, but would be nice.
